### PR TITLE
Fix FunctionApprovalResponseContent to message mapping 

### PR DESCRIPTION
**Pull request by PederHP**
When FunctionApprovalResponseContent items from the same original message (with MessageId = null) are processed, they incorrectly get split into separate FunctionCallContent messages instead of being grouped together.

Root Cause: Key mismatch in ConvertToFunctionCallContentMessages:
1. When creating messages from approvals, ConvertToFunctionCallContentMessage sets MessageId = fallbackMessageId if the original was null
2. Dictionary storage used currentMessage.MessageId (which is fallbackMessageId)
3. Dictionary lookup used RequestMessage?.MessageId ?? "" (which is "")
4. Keys don't match → messages aren't grouped

Fix (2 changes in FunctionInvokingChatClient.cs):
1. Line 1455-1458: When seeding the dictionary, use the effective key that accounts for fallbackMessageId substitution
2. Line 1464-1473: Use RequestMessage?.MessageId ?? "" consistently for both lookup and storage

This ensures all approvals from the same original message (whether MessageId is set or null) get properly grouped into a single FunctionCallContent message.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/7152)

### CODE REVIEW

**Comment by PederHP** [comment]
@dotnet-policy-service agree

**Comment by PederHP** [comment]
Fixes #7153

File: src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs [hunk 1]
```diff
@@ -1448,10 +1448,23 @@ private static (List<ApprovalResultWithRequestMessage>? approvals, List<Approval
...
                     messagesById = [];
-                    messagesById[currentMessage.MessageId ?? string.Empty] = currentMessage;
+
+                    // Use the effective key for the previous message, accounting for fallbackMessageId substitution.
+                    // If the message's MessageId was set to fallbackMessageId (because the original RequestMessage.MessageId was null),
+                    // we should use empty string as the key to match the lookup key used elsewhere.
+                    var previousMessageKey = currentMessage.MessageId == fallbackMessageId
+                        ? string.Empty
+                        : (currentMessage.MessageId ?? string.Empty);
+                    messagesById[previousMessageKey] = currentMessage;
```
[the remark this answers is not in the document]
Comment (PederHP) [reply src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs:1458]: GUID collision is an extremely theoretical edge case. Should I really add a comment documenting this?

File: test/Libraries/Microsoft.Extensions.AI.Tests/ChatCompletion/FunctionInvokingChatClientApprovalsTests.cs [hunk 2]
```diff
@@ -186,6 +186,69 @@ public async Task ApprovedApprovalResponsesAreExecutedAsync()
...
+    /// <summary>
+    /// Regression test: When messages containing FunctionApprovalRequestContent and FunctionApprovalResponseContent
+    /// have null MessageId, all approvals from the same original message should still be grouped into a single
+    /// FunctionCallContent message when sent to the downstream client.
+    ///
+    /// Previously, a bug in ConvertToFunctionCallContentMessages caused a key mismatch when MessageId was null:
+    /// - Messages were stored in a dictionary under the fallbackMessageId key
+    /// - But lookups used (RequestMessage?.MessageId ?? "") which resolved to ""
+    /// - This caused each FCC to be placed in a separate message instead of being grouped
+    /// </summary>
```
Comment (stephentoub) [inline test/Libraries/Microsoft.Extensions.AI.Tests/ChatCompletion/FunctionInvokingChatClientApprovalsTests.cs]:
```suggestion
```
